website/src/pages/Macro/Macro.jsx

```diff
@@ -0,0 +1,244 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import RingLoader from "react-spinners/RingLoader";
+import { Swiper, SwiperSlide } from "swiper/react";
+
+import '../Macro/macro.css'
+import "swiper/css";
+import "swiper/css/pagination";
+
+import { Mousewheel, Pagination, Keyboard } from "swiper";
+import Macrobench from '../../components/Macrobench/Macrobench';
+import MacrobenchMobile from '../../components/MacrobenchMobile/MacrobenchMobile';
+
+const Macro = () => {
+    const urlParams = new URLSearchParams(window.location.search);
+    const [dropDownLeft, setDropDownLeft] = useState(urlParams.get('ltag') == null ? 'Left' : urlParams.get('ltag'));
+    const [dropDownRight, setDropDownRight] = useState(urlParams.get('rtag') == null ? 'Right' : urlParams.get('rtag'));
+    const [openDropDownLeft, setOpenDropDownLeft] = useState(58);
+    const [openDropDownRight, setOpenDropDownRight] = useState(58);
+    const [dataRefs, setDataRefs] = useState([]);
+    const [dataMacrobench, setDataMacrobench] = useState([]);
+    const [commitHashLeft, setCommitHashLeft] = useState('')
+    const [commitHashRight, setCommitHashRight] = useState('')
+    const [error, setError] = useState(null);
+    const [isLoading, setIsLoading] = useState(true);
+    const [currentSlideIndex, setCurrentSlideIndex] = useState(urlParams.get('ptag') == null ? '0' : urlParams.get('ptag'));
+    const [currentSlideIndexMobile, setCurrentSlideIndexMobile] = useState(urlParams.get('ptagM') == null ? '0' : urlParams.get('ptagM'))
+    
+    
+    useEffect(() => {
+        const fetchData = async () => {
+          try {
+            const responseRefs = await fetch(`${import.meta.env.VITE_API_URL}vitess/refs`);
+      
+            const jsonDataRefs = await responseRefs.json();
+            
+            setDataRefs(jsonDataRefs);
+            
+            setIsLoading(false)
+          } catch (error) {
+            console.log('Error while retrieving data from the API', error);
+            setError('An error occurred while retrieving data from the API. Please try again.');
+            setIsLoading(false);
+          }
+        };
+      
+        fetchData();
+      }, []);
+
+    useEffect(() => {
+        const fetchData = async () => {
+            try {
+                const responseMacrobench = await fetch(`${import.meta.env.VITE_API_URL}macrobench/compare?rtag=${commitHashRight}&ltag=${commitHashLeft}`)
+
+                const jsonDataMacrobench = await responseMacrobench.json();
+
+                setDataMacrobench(jsonDataMacrobench)
+            } catch (error) {
+                console.log('Error while retrieving data from the API', error);
+                setError('An error occurred while retrieving data from the API. Please try again.');
+            }
+        };
+        fetchData();
```

Comment:
Did you make sure we don't do an unnecessary API call when both `dropDownLeft` and `dropDownRight` are `null`?